internal/controllers/machineset/machineset_controller_test.go

```diff
@@ -1326,6 +1328,193 @@ func TestMachineSetReconciler_syncMachines(t *testing.T) {
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "healthy-machine",
+				Namespace: "default",
+			},
+		}
+
+		machines := []*clusterv1.Machine{unhealthyMachine, healthyMachine}
+		fakeClient := fake.NewClientBuilder().WithObjects(controlPlaneUpgrading, unhealthyMachine, healthyMachine).WithStatusSubresource(&clusterv1.Machine{}).Build()
+		r := &Reconciler{Client: fakeClient}
+		_, err := r.reconcileUnhealthyMachines(ctx, cluster, machineSet, machines)
+		g.Expect(err).To(BeNil())
+		// Verify the unhealthy machine has the updated condition.
+		g.Eventually(func(g Gomega) {
+			m := &clusterv1.Machine{}
+			g.Expect(r.Client.Get(ctx, client.ObjectKeyFromObject(unhealthyMachine), m)).To(Succeed())
+			g.Expect(conditions.Has(m, clusterv1.MachineOwnerRemediatedCondition)).
+				To(BeTrue(), "Machine should have the %s condition set",
+					clusterv1.MachineOwnerRemediatedCondition)
+			machineOwnerRemediatedCondition := conditions.Get(m, clusterv1.MachineOwnerRemediatedCondition)
+			g.Expect(machineOwnerRemediatedCondition.Status).
+				To(Equal(corev1.ConditionFalse), "%s condition status should be false",
+					clusterv1.MachineOwnerRemediatedCondition)
+			g.Expect(machineOwnerRemediatedCondition.Reason).
+				To(Equal(clusterv1.WaitingForRemediationReason), "%s condition should have reason %s",
+					clusterv1.MachineOwnerRemediatedCondition, clusterv1.WaitingForRemediationReason)
+		}).Should(Succeed())
+	})
+}
+
+func TestMachineSetReconciler_syncReplicas(t *testing.T) {
+	t.Run("should hold off on creating new machines when preflight checks do not pass", func(t *testing.T) {
+		defer utilfeature.SetFeatureGateDuringTest(t, feature.Gates, feature.MachineSetPreflightChecks, true)()
+
+		g := NewWithT(t)
+
+		// An upgrading control plane should cause the preflight checks to not pass.
+		controlPlaneUpgrading := builder.ControlPlane("default", "test-cp").
+			WithVersion("v1.26.2").
+			WithStatusFields(map[string]interface{}{
+				"status.version": "v1.25.2",
+			}).
+			Build()
+		cluster := &clusterv1.Cluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-cluster",
+				Namespace: "default",
+			},
+			Spec: clusterv1.ClusterSpec{
+				ControlPlaneRef: contract.ObjToRef(controlPlaneUpgrading),
+			},
+		}
+		machineSet := &clusterv1.MachineSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-machineset",
+				Namespace: "default",
+			},
+			Spec: clusterv1.MachineSetSpec{
+				Replicas: pointer.Int32(1),
+			},
+		}
+
+		fakeClient := fake.NewClientBuilder().WithObjects(controlPlaneUpgrading, machineSet).WithStatusSubresource(&clusterv1.MachineSet{}).Build()
+		r := &Reconciler{Client: fakeClient}
+		result, err := r.syncReplicas(ctx, cluster, machineSet, nil)
+		g.Expect(err).To(BeNil())
+		g.Expect(result.IsZero()).To(BeFalse(), "syncReplicas should return a 'zero' result")
+		// Verify the proper condition is set on the MachineSet.
+		g.Expect(conditions.Has(machineSet, clusterv1.MachinesCreatedCondition)).
```

Comment:
No objection, but in my opinion those assertions are much more readable if we don't duplicate the assertion in the description text

Reply:
I am included to keeping them because incase of the test failures the message looks like:
```
Machine should have the OwnerRemediated condition set
Expected
   <bool>: true
to be false
```
instead of
```
Expected
    <bool>: true
 to be false
```
and the first one is more insightful.

Reply:
No objection :)